lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Server.hs

```diff
@@ -0,0 +1,44 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Test.Integration.Jormungandr.Scenario.CLI.Server
+    ( spec
+    ) where
+
+import Prelude
+
+import Control.Concurrent
+    ( threadDelay )
+import System.IO.Temp
+    ( withSystemTempDirectory )
+import System.Process
+    ( terminateProcess, withCreateProcess )
+import Test.Hspec
+    ( SpecWith, describe, it )
+import Test.Integration.Framework.DSL
+    ( Context (..), KnownCommand (..), expectPathEventuallyExist, proc' )
+
+spec :: forall t. KnownCommand t => SpecWith (Context t)
+spec = do
+    describe "SERVER - cardano-wallet serve" $ do
+        it "SERVER - Can start cardano-wallet serve --database" $ \_ -> do
+            withTempDir $ \d -> do
+                let db = d ++ "/db-file"
+                let args =
+                        [ "serve", "--database", db, "--genesis-hash"
+                        , "dba597bee5f0987efbf56f6bd7f44c38\
+                          \158a7770d0cb28a26b5eca40613a7ebd"
+                        ]
+                let process = proc' (commandName @t) args
+                withCreateProcess process $ \_ _ _ ph -> do
+                    expectPathEventuallyExist db
+                    expectPathEventuallyExist (db <> "-shm")
```

Comment:
I know this is copy&pasted from somewhere else, but will comment here anyway.

I don't think we need to assert that the `-shm` and `-wal` files exist. They are meant to be [temporary files](https://www.sqlite.org/tempfiles.html), and are a sqlite implementation detail. We just need to assert the db file was created.

Also if the `-wal` file exists after exit then that means we have not closed the database connection properly. Hmm.

Reply:
I asked a question in the past about this on a review (about what where those -shm and -wal expected to be). I agree that we could reduce this to test whether the db file exists as the rest seems more like an "implementation detail" from SQLite & Persistent.